staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -1364,6 +1365,22 @@ func HasSchemaWith(spec *apiextensions.CustomResourceDefinitionSpec, pred func(s
 	return false
 }
 
+var schemaPool = sync.Pool{
+	New: func() any {
+		return new(apiextensions.JSONSchemaProps)
```

Comment:
Will all of these objects be about the same size?

Reply:
Their size is known at compile time?

Reply:
I looked at the type and now I'm confused? There's a ton of pointers etc in it, are we clearing everything before storing it (or before reuse)?

Reply:
OK I actually looked down below and I'm surprised that actually avoids an allocation but if so it's pretty cool!